modules/rostests/apitests/user32/CharFuncs.c

```diff
@@ -0,0 +1,686 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for Char* functions
+ * COPYRIGHT:   Copyright 2022 Stanislav Motylkov <x86corez@gmail.com>
+ */
+
+#include "precomp.h"
+
+#include <winnls.h>
+#include <ndk/rtlfuncs.h>
+#include <pseh/pseh2.h>
+#include <strsafe.h>
+#include <versionhelpers.h>
+
+#define INVALID_PTR_OFF(x)  ((PVOID)(ULONG_PTR)(0xdeadbeefdeadbeefULL + x))
+#define INVALID_PTR         INVALID_PTR_OFF(0)
+#define TEST_ACP            1252
+
+typedef enum
+{
+    testLen,
+    testOffs,
+    testBoth,
+} TEST_TYPE;
+
+/* Dynamic allocation tests */
+typedef struct
+{
+    TEST_TYPE testType;
+    LPWSTR lpszStart;
+    LPWSTR lpszCurrent;
+    INT iOffset;
+    INT iResOffset;
+    INT iResOffsetNeg;
+    BOOL relStart;
```

Comment:
`relStart`
Is there any better name?

Reply:
Hmm, maybe `withinStart` (TRUE for pointer being within `lpszStart`, FALSE for within `lpszCurrent`).

Reply:
Please add a comment for it.